File: spec/cypress/integration/messenger/customization_spec.js
import {
  translations,
  openMessenger
} from '../../support/utils'


const defaultPredicates =  [
  {
    type: "match", 
    value: "and", 
    attribute: "match", 
    comparison: "and"
  }
]

describe('Customization Spec', function() {
  beforeEach(() => {
    cy.appEval('ActiveJob::Base.queue_adapter = :test')
    cy.appEval('ActiveJob::Base.queue_adapter.perform_enqueued_at_jobs = true')
  })

  it('display colourful', function() {
    cy.appScenario('basic')

    cy.appEval(`App.last.update(customization_colors: {
      primary: "#0f0",
      secondary: "#f00"
    })`)

    translations()
    openMessenger(($body)=>{
      cy.appEval("App.last").then((res)=>{
        expect(res.preferences.customization_colors).to.exist
      })
    }, {params: {lang: 'en'}} )
  })

  it('display default', function() {
    cy.appScenario('basic')

    cy.appEval(`App.last.update(customization_colors: nil )`)

    translations()
    openMessenger(($body)=>{
      cy.appEval("App.last").then((res)=>{
        expect(res.preferences.customization_colors).to.not.exist
      })
    }, {params: {lang: 'en'}} )
  })


  it('display default on secondary color', function() {
    cy.appScenario('basic')

    cy.appEval(`App.last.update(customization_colors: {
      primary: "#0f0",
      pattern: 'https://www.toptal.com/designers/subtlepatterns/patterns/spikes.png'
    })`)

    translations()
    openMessenger(($body)=>{
      cy.appEval("App.last").then((res)=>{
        expect(res.preferences.customization_colors).to.exist
      })
    }, {params: {lang: 'en'}} )
  })

})

File: spec/system/messenger/customization_spec.rb
require "rails_helper"

RSpec.describe "Customization Spec", type: :system, js: true do
  let!(:app) do
    Plugin.restore_plugins_from_fs

    FactoryBot.create(:app, encryption_key: "unodostrescuatro",
                            active_messenger: "true",
                            state: "enabled",
                            domain_url: "*")
  end

  before do
    ActiveJob::Base.queue_adapter = :test
    ActiveJob::Base.queue_adapter.perform_enqueued_at_jobs = true
    # Setup basic app scenario
    # ...
  end

  it "display colourful" do
    app.update(customization_colors: { primary: "#0f0", secondary: "#f00" })

    open_new_messenger({ params: { lang: "en" } }) do |_page|
      expect(app.customization_colors).to be_present
    end
  end

  it "display default" do
    app.update(customization_colors: nil)

    open_new_messenger({ params: { lang: "en" } }) do |_page|
      expect(app.customization_colors).to be_nil
    end
  end

  it "display default on secondary color" do
    app.update(customization_colors: { primary: "#0f0", pattern: "https://..." })

    open_new_messenger({ params: { lang: "en" } }) do |_page|
      expect(app.customization_colors).to be_present
    end
  end
end
